.github/workflows/maven.yml

```
@@ -55,16 +58,36 @@ jobs:
       with:
         name: liquibase-changelogs
         path: waltz-data/src/main/ddl/liquibase/
-    
+
     - name: Publish Test Report
       if: ${{ always() }}
       uses: scacap/action-surefire-report@v1
 
     - name: Release
       uses: softprops/action-gh-release@v1
-      if: startsWith(github.ref, 'refs/tags/')
+      if: ${{ github.ref_type == 'tag' }}
       with:
         files: |
           waltz-data/target/liquibase-scripts.zip
           waltz-web/target/waltz-web.war
           waltz-web/target/waltz-web-jar-with-dependencies.jar
+
+    - name: Login to GitHub Registry
+      if: ${{ github.ref_type == 'tag' }}
+      uses: docker/login-action@v1
+      with:
+        registry: ghcr.io
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: Build and push Docker image
+      if: ${{ github.ref_type == 'tag' }}
+      uses: docker/build-push-action@v2
+      with:
+        context: .
+        push: true
```

This will make it so that the Docker image is always built, but only pushed when it's a tag

```suggestion
      uses: docker/build-push-action@v2
      with:
        context: .
        push: ${{ github.ref_type == 'tag' }}
```

---

nice! thanks for this!